diff -cr bc-1.07.1/bc/fix-libmath_h bc-1.07.1-build.patch/bc/fix-libmath_h
*** bc-1.07.1/bc/fix-libmath_h	2017-04-08 01:20:02.000000000 +0300
--- bc-1.07.1-build.patch/bc/fix-libmath_h	2018-02-23 22:32:35.570536958 +0200
***************
*** 1,9 ****
! ed libmath.h <<EOS-EOS
! 1,1s/^/{"/
! 1,\$s/\$/",/
! 2,\$s/^/"/
! \$,\$d
! \$,\$s/,\$/,0}/
! w
! q
! EOS-EOS
--- 1,9 ----
! #! /bin/bash
! sed -e '1   s/^/{"/' \
!     -e     's/$/",/' \
!     -e '2,$ s/^/"/'  \
!     -e   '$ d'       \
!     -i libmath.h
! 
! sed -e '$ s/$/0}/' \
!     -i libmath.h
diff -cr bc-1.07.1/configure bc-1.07.1-build.patch/configure
*** bc-1.07.1/configure	2017-04-08 01:20:02.000000000 +0300
--- bc-1.07.1-build.patch/configure	2018-02-23 22:34:39.598464553 +0200
***************
*** 5288,5294 ****
     ?-?-flex)
  	LEX="flex -I -8" ;;
     ?-y-*)
! 	as_fn_error $? "readline works only with flex." "$LINENO" 5 ;;
  esac
  
  case $LEX-`uname -s` in
--- 5288,5294 ----
     ?-?-flex)
  	LEX="flex -I -8" ;;
     ?-y-*)
!     : ;; # as_fn_error $? "readline works only with flex." "$LINENO" 5 ;;
  esac
  
  case $LEX-`uname -s` in
